analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/LooseFilePermissions.cs

```diff
@@ -0,0 +1,152 @@
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, identifier.GetLocation()));
+            }
+        }
+
+        private static bool IsFileAccessPermissions(SimpleNameSyntax identifierNameSyntax, SemanticModel semanticModel) =>
+            WeakFileAccessPermissions.Contains(identifierNameSyntax.Identifier.Text)
+            && identifierNameSyntax.IsKnownType(KnownType.Mono_Unix_FileAccessPermissions, semanticModel);
+
+        private static void VisitInvocations(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (InvocationExpressionSyntax)context.Node;
+            if (!invocation.IsMemberAccessOnKnownType("SetAccessRule", KnownType.System_Security_AccessControl_FileSystemSecurity, context.SemanticModel) &&
+                !invocation.IsMemberAccessOnKnownType("AddAccessRule", KnownType.System_Security_AccessControl_FileSystemSecurity, context.SemanticModel))
+            {
+                return;
+            }
+
+            if (GetObjectCreation(invocation, context.SemanticModel) is { })
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, invocation.GetLocation()));
+            }
+        }
+
+        private static ObjectCreationExpressionSyntax GetObjectCreation(InvocationExpressionSyntax invocation, SemanticModel semanticModel)
+        {
+            var accessRuleSyntaxNode = GetVulnerableFileSystemAccessRule(invocation.DescendantNodes());
+            if (accessRuleSyntaxNode != null)
+            {
+                return accessRuleSyntaxNode;
+            }
+
+            var accessRuleSymbol = invocation.GetArgumentSymbolsOfKnownType(KnownType.System_Security_AccessControl_FileSystemAccessRule, semanticModel).FirstOrDefault();
+            if (accessRuleSymbol == null || accessRuleSymbol is IMethodSymbol)
+            {
+                return null;
+            }
+
+            return GetVulnerableFileSystemAccessRule(accessRuleSymbol.GetLocationNodes(invocation));
+
+            ObjectCreationExpressionSyntax GetVulnerableFileSystemAccessRule(IEnumerable<SyntaxNode> nodes) =>
+                nodes.OfType<ObjectCreationExpressionSyntax>()
+                     .FirstOrDefault(objectCreation => IsFileSystemAccessRuleForEveryoneWithAllow(objectCreation, semanticModel));
+        }
+
+        private static bool IsFileSystemAccessRuleForEveryoneWithAllow(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel) =>
+            objectCreation.IsKnownType(KnownType.System_Security_AccessControl_FileSystemAccessRule, semanticModel)
+            && objectCreation.ArgumentList is { } argumentList
+            && IsEveryone(argumentList.Arguments.First().Expression, semanticModel)
+            && semanticModel.GetConstantValue(argumentList.Arguments.Last().Expression) is {HasValue: true, Value: 0};
+
+        private static bool IsEveryone(SyntaxNode syntaxNode, SemanticModel semanticModel)
```

Comment:
Styling: If could be converted to conditional expression and expression bodied method